client/src/lib/store/track.ts

```diff
@@ -88,13 +88,24 @@ export const useTrack = create<TrackState>()((set) => ({
     return { success: true, message: "Application added successfully!" };
   },
 
-  removeApplication: (from, applicationId) =>
+  removeApplication: async (from, application, profile): Promise<{success: boolean; message: string}> => {
     set((state) => {
       state[from] = state[from].filter(
-        (app) => app.application_id !== applicationId
+        (app) => app.application_id !== application.application_id
       );
       return { ...state };
-    }),
+    });
+    
+    // Call the supabase function to remove the application
+    const {error } = await RemoveApplication(profile, application.application_id);
```

Comment:
I would only delete the application from the store if the supabase operation is successful first. This prevents inconsistency with the UI if the supabase operation fails

Reply:
okie, that makes sense. I will fix the logic